deps/z3/Makefile

```diff
@@ -0,0 +1,20 @@
+Z3_VERS=4.8.9
+Z3_URL_PREFIX=https://github.com/Z3Prover/z3/releases/download/z3-$(Z3_VERS)
+Z3_BASE_NAME=z3-${Z3_VERS}-x64-ubuntu-16.04
```

Comment:
Like Isabelle, this is not going to work on MacOS and Windows.

Reply:
I have updated the makefiles (for Isabelle and other backends). All work on MacOS and Cygwin except Isabelle:
  - On MacOS, I'm getting `poly: Bad CPU type in executable`, I guess the 32 bit support is missing. At least it was needed on Linux for Isabelle to work.
  - On Cygwin, the following DLLs/Versions are missing: `cyggcc_s-1.dll`, `cyggmp-3.dll`. The latter was added in the old release script, but the DLL is not committed to the repo, thus I guess it should fail as well.